File: app/src/main/java/com/snowdango/numac/AppDataEditor/AppDataEditorFragment.java
package com.snowdango.numac.AppDataEditor;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.text.InputFilter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.snowdango.numac.ListFormat.AppListFormat;
import com.snowdango.numac.R;

import static com.snowdango.numac.NumAcMain.NumAcActivity.list;
import static com.snowdango.numac.NumAcMain.NumAcActivity.metrics;

public class AppDataEditorFragment extends Fragment implements View.OnClickListener {

    private Button[] buttons;
    public TextView appNameView;
    private EditText appCommand;
    public String appName = "Error";
    public String oldCommand = "0000";
    private int appPosition = -1;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        appName = getActivity().getIntent().getStringExtra("appName");
        AppDataEditorActivity appDataEditorActivity = new AppDataEditorActivity();
        appPosition = appDataEditorActivity.searchAppFormList(appName);
        if(appPosition != -1) {
            AppListFormat appData = list.get(appPosition);
            oldCommand = appData.getAppCommand();
        }
        return inflater.inflate(R.layout.fragment_appdata,container,false);
    }

    @Override
    public void onViewCreated(@NonNull View v, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(v, savedInstanceState);

        InputFilter[] filters = new InputFilter[]{new InputFilter.LengthFilter(4)};
        appCommand = v.findViewById(R.id.target_app_command);
        appCommand.setFilters(filters);
        appCommand.setText(oldCommand);

        appNameView = v.findViewById(R.id.target_app_name);
        appNameView.setPadding(0,metrics.heightPixels/4,0,9*0);
        appNameView.setText(appName);

        buttons = new Button[]{v.findViewById(R.id.uninstall_button),v.findViewById(R.id.replace_button)};
        for(Button b : buttons) {
            b.setWidth(metrics.widthPixels / 3);
            b.setHeight(metrics.heightPixels / 14);
            b.setOnClickListener(this);
        }

        if(appPosition == -1){
            ClickButtonEvents c = new ClickButtonEvents();
            c.AlertCreate("Sorry","Sorry, Can't find this app.\n" +
                            "please push reload appList command(\"##00\")",
                    "OK",null,null,null);
        }
    }

    @Override
    public void onClick(View v) {
        ClickButtonEvents c = new ClickButtonEvents();
        if(appPosition != -1) {
            switch (v.getId()) {
                case R.id.uninstall_button:
                    c.AlertCreate("UnInstallApp","Really? Do you want yo uninstall this app?","OK"
                            ,(DialogInterface dialog, int which) -> {
                                Intent uninstallIntent = c.uninstallApp(oldCommand);
                                if (uninstallIntent != null) {
                                    startActivity(uninstallIntent);
                                    c.deleteApp(appPosition, appName, getActivity());
                                }else{
                                    c.AlertCreate("Sorry", "I can't  this app.", "OK"
                                            , null, null, null);
                                }
                            },"Cancel" , null);
                    break;
                case R.id.replace_button:
                    c.AlertCreate("Change Command", "Really? Do you want yo change this app's command ?" ," OK"
                            , (DialogInterface dialog, int which) -> {
                                String newCommand = appCommand.getText().toString();
                                if (c.checkCommandFormat(newCommand)) {
                                    if (c.checkNewCommandForList(newCommand)) {
                                        if (c.changeAppCommand(appPosition, appName, newCommand)) {
                                            c.AlertCreate("Finish", "Finish update!! \n I change this app's command"
                                                    , "OK", null, null, null);
                                        }
                                    }
                                }
                            }, "Cancel" , null ) ;
                    break;
            }
        }else{
            c.AlertCreate("Sorry","Sorry, Can't find this app.\n" +
                            "please push reload appList command(\"##00\")",
                    "OK",null,null,null);
        }
    }
}

File: app/src/main/java/com/snowdango/numac/AppDataEditor/AppDataEditorFragment.kt
@file:Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")

package com.snowdango.numac.AppDataEditor

import android.content.DialogInterface
import android.os.Bundle
import android.text.InputFilter
import android.text.InputFilter.LengthFilter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.snowdango.numac.NumAcMain.NumAcActivity
import com.snowdango.numac.R

class AppDataEditorFragment : Fragment(), View.OnClickListener {
    private lateinit var buttons: Array<Button>
    lateinit var appNameView: TextView
    private lateinit var appCommand: EditText
    var appName = "Error"
    var oldCommand = "0000"
    private var appPosition = -1
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        appName = activity!!.intent.getStringExtra("appName")
        val appDataEditorActivity = AppDataEditorActivity()
        appPosition = appDataEditorActivity.searchAppFormList(appName)
        if (appPosition != -1) {
            val appData = NumAcActivity.list!![appPosition]
            oldCommand = appData.appCommand.toString()
        }
        return inflater.inflate(R.layout.fragment_appdata, container, false)
    }

    override fun onViewCreated(v: View, savedInstanceState: Bundle?) {
        super.onViewCreated(v, savedInstanceState)
        val filters = arrayOf<InputFilter>(LengthFilter(4))
        appCommand = v.findViewById(R.id.target_app_command)
        appCommand.filters = filters
        appCommand.setText(oldCommand)
        appNameView = v.findViewById(R.id.target_app_name)
        appNameView.setPadding(0, NumAcActivity.metrics!!.heightPixels / 4, 0, 9 * 0)
        appNameView.text = appName
        buttons = arrayOf(v.findViewById(R.id.uninstall_button), v.findViewById(R.id.replace_button))
        for (b in buttons) {
            b.width = NumAcActivity.metrics!!.widthPixels / 3
            b.height = NumAcActivity.metrics!!.heightPixels / 14
            b.setOnClickListener(this)
        }
        if (appPosition == -1) {
            val c = ClickButtonEvents()
            c.AlertCreate("Sorry", """Sorry, Can't find this app.please push reload appList command("##00")""".trimIndent(),
                    "OK", null, null, null)
        }
    }

    override fun onClick(v: View) {
        val c = ClickButtonEvents()
        if (appPosition != -1) {
            when (v.id) {
                R.id.uninstall_button -> c.AlertCreate("UnInstallApp", "Really? Do you want yo uninstall this app?", "OK"
                        , DialogInterface.OnClickListener { _: DialogInterface?, _: Int ->
                    val uninstallIntent = c.uninstallApp(oldCommand)
                    if (uninstallIntent != null) {
                        startActivity(uninstallIntent)
                        c.deleteApp(appPosition, appName, activity)
                    } else {
                        c.AlertCreate("Sorry", "I can't  this app.", "OK"
                                , null, null, null)
                    }
                }, "Cancel", null)
                R.id.replace_button -> c.AlertCreate("Change Command", "Really? Do you want yo change this app's command ?", " OK"
                        , DialogInterface.OnClickListener { _: DialogInterface?, _: Int ->
                    val newCommand = appCommand.text.toString()
                    if (c.checkCommandFormat(newCommand)) {
                        if (c.checkNewCommandForList(newCommand)) {
                            if (c.changeAppCommand(appPosition, appName, newCommand)) {
                                c.AlertCreate("Finish", "Finish update!! \n I change this app's command"
                                        , "OK", null, null, null)
                            }
                        }
                    }
                }, "Cancel", null)
            }
        } else {
            c.AlertCreate("Sorry", """Sorry, Can't find this app.please push reload appList command("##00")""".trimIndent(),
                    "OK", null, null, null)
        }
    }
}